clients/include/testing_trsm_strided_batched.hpp

```diff
@@ -0,0 +1,453 @@
+            {
+                T err_1 = 0.0;
+                T err_2 = 0.0;
+                for(int j = 0; j < M; j++)
+                {
+                    int idx = j + i * ldb + b * stride_b;
+                    if(hX[idx] != 0)
+                    {
+                        err_1 += std::abs((hX[idx] - hXorB_1[idx]) / hX[idx]);
+                        err_2 += std::abs((hX[idx] - hXorB_2[idx]) / hX[idx]);
+                    }
+                    else
+                    {
+                        err_1 += std::abs(hXorB_1[idx]);
+                        err_2 += std::abs(hXorB_2[idx]);
+                    }
+                }
+                max_err_1 = max_err_1 > err_1 ? max_err_1 : err_1;
+                max_err_2 = max_err_2 > err_2 ? max_err_2 : err_2;
+            }
+            trsm_err_res_check<T>(max_err_1, M, error_eps_multiplier, eps);
+            trsm_err_res_check<T>(max_err_2, M, error_eps_multiplier, eps);
+        }
+
+        // Residual Check
+        // hXorB <- hA * (A^(-1) B) ;
+        for(int b = 0; b < batch_count; b++)
+        {
+            cblas_trmm<T>(side,
+                          uplo,
+                          transA,
+                          diag,
+                          M,
+                          N,
+                          1.0 / alpha_h,
+                          hA + stride_a * b,
+                          lda,
+                          hXorB_1 + stride_b * b,
+                          ldb);
+            cblas_trmm<T>(side,
+                          uplo,
+                          transA,
+                          diag,
+                          M,
+                          N,
+                          1.0 / alpha_h,
+                          hA + stride_a * b,
+                          lda,
+                          hXorB_2 + stride_b * b,
+                          ldb);
+        }
+
+        // hXorB contains A * (calculated X), so residual = A * (calculated X) - B
+        //                                                = hXorB - hB
+        // res is the one norm of the scaled residual for each column
+        for(int b = 0; b < batch_count; b++)
+        {
+            max_res_1 = max_res_2 = 0;
+            for(int i = 0; i < N; i++)
+            {
+                T res_1 = 0.0;
+                T res_2 = 0.0;
+                for(int j = 0; j < M; j++)
+                {
+                    int idx = j + i * ldb + b * stride_b;
+                    if(hB[j + i * ldb] != 0)
+                    {
+                        res_1 += std::abs((hXorB_1[idx] - hB[idx]) / hB[idx]);
+                        res_2 += std::abs((hXorB_2[idx] - hB[idx]) / hB[idx]);
+                    }
+                    else
+                    {
+                        res_1 += std::abs(hXorB_1[idx]);
+                        res_2 += std::abs(hXorB_2[idx]);
+                    }
+                }
+                max_res_1 = max_res_1 > res_1 ? max_res_1 : res_1;
```

Comment:
I'm sure using std::max wouldn't be a problem, I can go ahead and change it, I'm guessing this logic is used elsewhere as well, though.

Reply:
> 
> 
> I'm sure using std::max wouldn't be a problem, I can go ahead and change it, I'm guessing this logic is used elsewhere as well, though.

`std::` functions cannot generally be used in `__device__` code (and thus `__global__` kernels).

`std::abs` was replaced with `rocblas_abs`. If `max` is needed, then a `rocblas_max` should be added to `library/src/include/utility.h`

Reply:
This is host/testing code so std:: functions can be used. I'm not going to change it here as we use this logic in quite a few other files for testing; this PR is big enough. If we want to change these to std::max (or otherwise) we should do another refactoring PR I think.